css_variable/index-START.html

```diff
@@ -0,0 +1,75 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>Scoped CSS Variables and JS</title>
+</head>
+<body>
+  <h2>Update CSS Variables with <span class='hl'>JS</span></h2>
+
+  <div class="controls">
+    <label for="spacing">Spacing:</label>
+    <input id="spacing" type="range" name="spacing" min="10" max="200" value="10" data-sizing="px">
+
+    <label for="blur">Blur:</label>
+    <input id="blur" type="range" name="blur" min="0" max="25" value="10" data-sizing="px">
+
+    <label for="base">Base Color</label>
+    <input id="base" type="color" name="base" value="#3700ff">
+  </div>
+
+  <img src="sunset.png" width="600">
+
+  <style>
+    :root{
+      --base: #3700ff;
+      --spacing: 10px;
+      --blur: 10px;
+    }
+
+    img{
+      padding: var(--spacing);
+      background: var(--base);
+      filter:blur(var(--blur));
+    }
+
+    .hl{
+      color: var(--base);
+    }
+    /*
+      misc styles, nothing to do with CSS variables
+    */
+
+    body {
+      text-align: center;
+      background: #193549;
+      color: white;
+      font-family: 'helvetica neue', sans-serif;
+      font-weight: 100;
+      font-size: 50px;
+    }
+
+    .controls {
+      margin-bottom: 50px;
+    }
+
+    input {
+      width: 100px;
+    }
+  </style>
+
+  <script>
+    const input = document.querySelector('.controls');
+
+    function handleUpdate(e){
+      if (e.target.tagName !== "INPUT") return;
+      const suffix = e.target.dataset.sizing || ""
+      document.documentElement.style.setProperty(`--${e.target.name}`,e.target.value + suffix)
+    }
+
+    input.addEventListener('change', handleUpdate);
```

Comment:
Using a single event handler on multiple element type is an anti-pattern.

Reply:
okay